### ISSUE TITLE
Unexpected YAML multline string handling in task commands

### ISSUE THREAD

**Issue by hnicke** [garden-io/garden#4807]
## Bug

Multiline yaml strings in `tasks[].command[]` are handled differently than expected.

### Current Behavior

A multiline string is not handled as one argument, but as multiple, since the shell tokenization splits the multiline yaml string into multiple arguments, using whitespace as delimiter.

### Expected behavior

A YAML multiline string should be handled as a single argument.
I.e., the mechanism should work just like kubernetes handles `spec.containers[].command` or `spec.containers[].args` field.

In kubernetes, it is a common pattern to use inline shell scripts in multiline yaml strings like this:
```yaml
kind: Pod
...
containers:
  - command:
      - bash
      - -c
      - |
        set -eu
        echo this is my inline shell script
```

In garden tasks, extra quoting around the multiline string is required.
This behavior is very confusing to me.
It requires to escape the quotes in the shell script, which gets super confusing quickly.
Additionally, it is quite error prone: It happened multiple times to me that I accidentally included an unescaped quote character in the multiline string, which led to unintended outcomes:

```yaml
kind: Module
type: exec
name: example
tasks:
  - name: example
    command:
      - bash
      - -c
      - |
        'set -eu
        # long and tedious script here
        # then, I'm adding a quote << did you see it
        # my script is broken from here on
        '
```

### Reproducible example

```yaml
kind: Module
type: exec
name: example
tasks:
  - name: example
    command:
      - echo
      - |
        line 1
        line 2
```

```
$ garden run example -l debug

...
line 1
/bin/sh: line 2: line: command not found
Command failed with exit code 127: echo line 1
line 2
```

### Workaround

Quote the multiline string:
```yaml
kind: Module
type: exec
name: example
tasks:
  - name: example
    command:
      - echo
      - |
        'line 1
        line 2'
```

```
garden run example -l debug

...
line 1
line 2
```

### Additional context

Changing this is a breaking change but I think it's worth it.

### Your environment

* **OS:** arch linux
* **How I'm running Kubernetes:** GKE

`garden version`
0.13.7, but has been like this since a long time

**Comment by hnicke** [comment]
This issue seems to only occur with Modules. Actions properly handle this scenario.
It's probably not worth it to adjust the behavior of Modules anymore.

**Comment by 10ko** [comment]
As suggested above, the behavior has been fixed with `0.13.x` actions.
